api/chaos/serverexperimentation/v1/serverexperimentation.proto

```diff
@@ -56,3 +56,128 @@ enum FaultInjectionCluster {
   // Faults are injected to ingesss traffic by an upstream service(s).
   FAULTINJECTIONCLUSTER_UPSTREAM = 2;
 }
+
+// The configuration of an HTTP fault.
+message HTTPFaultConfig {
+  oneof fault {
+    option (validate.required) = true;
+
+    // The abort fault.
+    AbortFault abort_fault = 1;
+    // The latency fault.
+    LatencyFault latency_fault = 2;
+  }
+}
+
+// The definition of an abort fault.
+message AbortFault {
+  // The targeting of the fault describing what requests faults should be applied to.
+  FaultTargeting fault_targeting = 1 [ (validate.rules).message.required = true ];
+  // The percentage of requests the fault should be applied to.
+  FaultPercentage fault_parcentage = 2 [ (validate.rules).message.required = true ];
+  // The abort status to apply.
+  FaultAbortStatus fault_abort_status = 3 [ (validate.rules).message.required = true ];
+}
+
+// The definition of a latency fault.
+message LatencyFault {
+  // The targeting of the fault describing where the faults should be applied.
+  FaultTargeting fault_targeting = 1 [ (validate.rules).message.required = true ];
+  // The percentage of requests the fault should be applied to.
+  FaultPercentage fault_parcentage = 2 [ (validate.rules).message.required = true ];
+  // The latency duration to apply.
+  FaultLatencyDuration fault_latency_duration = 3 [ (validate.rules).message.required = true ];
+}
+
+// The fault targeting that describes a part of the system that's responsible for enforcing a fault.
```

Comment:
This could use some high level explanation about why this structure is used; how it allows us to support various forms of targeting criteria based on whether we're having the upstream or the downstream enforce the faults.